main.go

```diff
@@ -118,6 +129,8 @@ func main() {
 		accessLogger := log.New(os.Stdout, "", log.Ldate|log.Ltime|log.LUTC)
 		errorLogger := log.New(os.Stderr, "", log.Ldate|log.Ltime|log.LUTC)
 
+		writePidFile(c, accessLogger)
```

Comment:
I find it surprising that a function called `writePidFile` would have the side-effect of killing other processes. Better to decouple that logic.

Reply:
agreed